TombEngine/Objects/TR3/Entity/tr3_civvy.cpp

```diff
@@ -2,418 +2,601 @@
+					}
+
+					if (creature.Flags == 2)
+					{
+						DoDamage(creature.Enemy, CIVVY_ATTACK_DAMAGE);
+						CreatureEffect(&item, CivvyBiteLeft, DoBloodSplat);
+						SoundEffect(SFX_TR4_LARA_THUD, &item.Pose);
+						creature.Flags = 1;
+
+						if (creature.Enemy->HitPoints <= 0 && !creature.Enemy->IsLara() && !item.HitStatus)
+							creature.Alerted = false;
+					}
 				}
 
 				break;
 
-			case CIVVY_PUNCH1:
-				creature->MaxTurn = CIVVY_WALK_TURN_RATE_MAX;
+			case CIVVY_STATE_ATTACK_FAR_PUNCH:
+
+				creature.MaxTurn = CIVVY_WALK_TURN_RATE_MAX;
 
 				if (AI.ahead)
 				{
-					extraTorsoRot.x = AI.xAngle;
-					extraTorsoRot.y = AI.angle;
+					jointTorsoRot.x = AI.xAngle;
+					jointTorsoRot.y = AI.angle;
 				}
 
-				if (!creature->Flags && item->TouchBits.Test(CivvyAttackJoints))
+				if (creature.Flags == 0)
 				{
-					CreatureEffect(item, CivvyBite, DoBloodSplat);
-					DoDamage(creature->Enemy, CIVVY_ATTACK_DAMAGE);
-					SoundEffect(SFX_TR4_LARA_THUD, &item->Pose);
-					creature->Flags = 1;
+					if (creature.Enemy->IsLara())
+					{
+						if (item.TouchBits.Test(CivvyAttackJoints))
+							creature.Flags = 2;
+					}
+					else
+					{
+						float distance = Vector3i::Distance(item.Pose.Position, creature.Enemy->Pose.Position);
+						if (distance <= CLICK(2))
+							creature.Flags = 2;
+					}
+
+					if (creature.Flags == 2)
+					{
+						DoDamage(creature.Enemy, CIVVY_ATTACK_DAMAGE);
+						CreatureEffect(&item, CivvyBiteLeft, DoBloodSplat);
+						SoundEffect(SFX_TR4_LARA_THUD, &item.Pose);
+						creature.Flags = 1;
+
+						if (creature.Enemy->HitPoints <= 0 && !creature.Enemy->IsLara() && !item.HitStatus)
+							creature.Alerted = false;
+					}
 				}
 
-				if (AI.ahead && AI.distance > CIVVY_ATTACK1_RANGE && AI.distance < CIVVY_ATTACK2_RANGE)
-					item->Animation.TargetState = CIVVY_PUNCH2;
+				if (AI.ahead && AI.distance > CIVVY_ATTACK_FAR_PUNCH_RANGE)
+					item.Animation.TargetState = CIVVY_STATE_WALK;
 
 				break;
 
-			case CIVVY_PUNCH2:
-				creature->MaxTurn = CIVVY_WALK_TURN_RATE_MAX;
+			case CIVVY_STATE_ATTACK_WALKING_PUNCH:
+
+				creature.MaxTurn = CIVVY_WALK_TURN_RATE_MAX;
 
 				if (AI.ahead)
 				{
-					extraTorsoRot.x = AI.xAngle;
-					extraTorsoRot.y = AI.angle;
+					jointTorsoRot.x = AI.xAngle;
+					jointTorsoRot.y = AI.angle;
 				}
 
-				if (creature->Flags != 2 && item->TouchBits.Test(CivvyAttackJoints))
+				if (creature.Flags == 0)
 				{
-					DoDamage(creature->Enemy, CIVVY_SWIPE_DAMAGE);
-					CreatureEffect(item, CivvyBite, DoBloodSplat);
-					SoundEffect(SFX_TR4_LARA_THUD, &item->Pose);
-					creature->Flags = 2;
+					if (creature.Enemy->IsLara())
+					{
+						if (item.TouchBits.Test(CivvyAttackJoints))
+							creature.Flags = 2;
+					}
+					else
+					{
+						float distance = Vector3i::Distance(item.Pose.Position, creature.Enemy->Pose.Position);
+						if (distance <= CLICK(2))
+							creature.Flags = 2;
+					}
+
+					if (creature.Flags == 2)
```

Comment:
Probably good idea is to demagic `creature.Flags` meanings to an enum.

Reply:
Done.

Although I think in the future when the damage refactor get done, it will be useful make a similar enum there that can be used by all enemies that do melee attacks.